launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -46,24 +47,15 @@ static boolean isEmpty(String s) {
 
   /** Joins a list of strings using the given separator. */
   static String join(String sep, String... elements) {
-    StringBuilder sb = new StringBuilder();
-    for (String e : elements) {
-      if (e != null) {
-        if (sb.length() > 0) {
-          sb.append(sep);
-        }
-        sb.append(e);
-      }
-    }
-    return sb.toString();
+    return join(sep, Arrays.stream(elements).toList());
```

Comment:
IIRC, in relatively simple computing scenarios, `Arrays.stream` is expected to be slower than the existing implementation. Has its performance improved now?

Reply:
fine for me, but, please update the PR title to better reflect what this PR has done.